model-view-viewmodel/pom.xml

```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<artifactId>java-design-patterns</artifactId>
+		<groupId>com.iluwatar</groupId>
+		<version>1.24.0-SNAPSHOT</version>
+	</parent>
+	<groupId>com.iluwatar</groupId>
+	<artifactId>model-view-viewmodel</artifactId>
+	<version>1.24.0-SNAPSHOT</version>
+	<properties>
+		<zk.version>9.0.0</zk.version>
+		<guava.version>19.0</guava.version>
+		<jetty-maven-plugin.version>9.4.28.v20200408</jetty-maven-plugin.version>
+		<maven-war-plugin.version>2.1.1</maven-war-plugin.version>
+		<maven-assembly-plugin.version>2.2</maven-assembly-plugin.version>
+		<maven.build.timestamp.format>yyyy-MM-dd</maven.build.timestamp.format>
+		<packname>-${project.version}-FL-${maven.build.timestamp}</packname>
+	</properties>
+	<packaging>war</packaging>
+	<name>model-view-viewmodel</name>
+	<description>model-view-viewmodel</description>
+	<licenses>
+		<license>
+			<name>GNU LESSER GENERAL PUBLIC LICENSE, Version 3</name>
+			<url>https://www.gnu.org/licenses/lgpl.html</url>
+			<distribution>repo</distribution>
+		</license>
+	</licenses>
```

Comment:
Having a separate LGPL licensed module in the project might be problematic. I think we need to look into this and discuss it a bit more.

Reply:
See this link: https://www.zkoss.org/license

Reply:
Is the pattern licensed using GPL3 or is it the library zkoss is?

@jinishavora

Reply:
> See this link: https://www.zkoss.org/license

So we can use it over here, but as per GPL I believe we need to leave it as it is.

Reply:
I believe having this new LGPL-licensed module as a dependency would make our project what they 'combined work' in https://www.gnu.org/licenses/lgpl-3.0.html and we would have to fulfill those requirements.

Reply:
Well we are not modifying any part of the framework, but using it. So I guess we are okay including it in the pom and be done with it right?

Reply:
If the sample application using the framework really is LGPL licensed, I think it's a bit more complicated than that. We would at least have to

- [ ] Give prominent notice with each copy of the Combined Work that the Library is used in it and that the Library and its use are covered by this License.
- [ ] Accompany the Combined Work with a copy of the GNU GPL and this license document.
- [ ] Configure license-maven-plugin so that it puts a different license to the files under model-view-viewmodel

Reply:
So this module need to follow the LGPL license while the rest of the project will continue with MIT?

Reply:
Yes, that's at least how I understand it. I think the above criteria should be fulfilled by this pull request. An alternative approach would be getting rid of zkoss framework and creating the pattern without it or with something else.

Reply:
One consideration is also how this affects our "end-users". Currently, everything is published under the permissive MIT license. I'm not strict with it, but if one reads and interprets it very carefully and literally, copying and pasting code from this repository to some other project would require the copier to MIT license the copied portion (see for example the discussion [here](https://github.com/salesforce/gorp/issues/2)). If we include this LGPL licensed module, it becomes more confusing from the end user's point of view due to different licensing rules depending on where you take the code from.

Reply:
What's the verdict here? Do we want to keep this as is or not use zkoss and find something else with MIT license.
I am not really sure if we can find something for this pattern under MIT license.

Reply:
I'm thinking that we can merge the pattern as is and fulfill the LGPL licensing requirements in a follow-up ticket. It's a permissive license anyway, we just need to make some adjustments here and there.

Reply:
Sounds good!
Once approved, I can merge this and then make the changes required in the follow-up.

Reply:
Sounds like a good idea.

Reply:
Follow-up licensing task: https://github.com/iluwatar/java-design-patterns/issues/1690